### PR TITLE
Implement ``zigup default master``

### LINKED ISSUE #33: zig default master
The `zig default` command is meant to set the default compiler without fetching it.  In contrast, when referencing the zig `master` version, the latest version is requested from the zig download site each time.  This would mean that `zigup default master` would fail with "compiler not installed" if the website has published a new version since it was last fetched.  That being said, the "spirit" of the default command is not to fetch anything, so in this case, I think it makes more sense to have `zigup default master` set the default compiler to the last master version that was fetched, rather than trying to get the latest version and fail most of the time.

### PR DESCRIPTION
**Pull request by iddev5**
It just removes the explicit check as missing of ``master`` symlink would already get detected by setDefaultCompiler().
Implements as per discussed on discord/issue topic.

I noticed that you aren't doing zig fmt, so I didn't did it either.

Closes #33

### CODE REVIEW

File: zigup.zig [hunk 1]
```diff
@@ -292,11 +292,7 @@ pub fn main2() !u8 {
             defer allocator.free(install_dir);
             const compiler_dir = try std.fs.path.join(allocator, &[_][]const u8{ install_dir, version_string });
             defer allocator.free(compiler_dir);
-            if (std.mem.eql(u8, version_string, "master")) {
-                @panic("set default to master not implemented");
-            } else {
-                try setDefaultCompiler(allocator, compiler_dir, .verify_existence);
-            }
+            try setDefaultCompiler(allocator, compiler_dir, .verify_existence);
```
Comment (marler8997) [inline zigup.zig:295]: I'm not sure if this is correct or not.  Could you add a test for this (in `test.zig`)?

Comment (iddev5) [reply zigup.zig:295]: I have been using it since the day I created this PR, it is working though. Not sure about any hidden edge cases, but I did tested it a bit then and no problem found, afterall master is effectively a symlink to something.

Anyways, I will try to add a test once I figure out how it works.

Comment (iddev5) [reply zigup.zig:295]: I tried to add a test case here: https://github.com/marler8997/zigup/pull/34/commits/ab2413082b023f11529b1c1d1da944b31416f49b 

Couldn't figure out a better way since there's nothing to check in the output

**Comment by iddev5** [comment]
I have also rebased it to master

File: zigup.zig (hunk 1 above)
Comment (marler8997) [inline zigup.zig:295]: I don't think this will work on windows since it doesn't have symlinks (not sure why github actions isn't testing your PR).  I'm fairly certain that what needs to be done is when we see "master", we need to resolve the master version (if master has not been downloaded print a nice error that it hasn't been fetched), and then call setDefaultCompiler with that specific version.

Comment (marler8997) [reply zigup.zig:295]: Also we should test the case where we run `zigup deafult master` before we have downloaded anything and verify we get a good error message.

If you want me to finish the PR I can but I'll give you a change to implement it if you like.

Comment (iddev5) [reply zigup.zig:295]: Oh alright. I don't have a windows PC so I couldn't test it and thus I cannot reliably implement it either. So yeah please continue.

**Comment by marler8997** [comment]
Ok I finished the implementation and added a test to make sure that we get a good error when master is not fetched.  I'll give you a change to review the code before merging.

**Comment by iddev5** [comment]
Ohh it makes sense now, looks good
